src/AppInstallerRepositoryCore/CompositeSource.cpp

```diff
@@ -701,26 +764,34 @@ namespace AppInstaller::Repository
             }
 
             // Gets the information about the pins that exist for this package
-            void GetExistingPins(PinningIndex& pinningIndex, bool cleanUpStalePins)
+            void GetExistingPins(PinningIndex& pinningIndex)
             {
-                // If the package is installed, we need to add the pin information to the available packages from any source.
-                // If the package is not installed, we clean up stale pin information here.
+                if (!m_installedPackage)
+                {
+                    // Only installed packages have pins

```

Comment:
Does not seem the case any more if pin by id where we do not ensure an installed version found

Reply:
Updated the comment. A package that is not installed can have pins, but we don't care about them. For pin operations in those cases we bypass the composite source, and pins don't affect packages that are not installed

Reply:
I feel we could just remove this check and always add pin info to the composite package, for correctness

Reply:
Removed